pkg/kp/kv.go

```diff
@@ -240,6 +241,35 @@ func (c consulStore) ListPods(keyPrefix string) ([]ManifestResult, time.Duration
 	return ret, queryMeta.RequestTime, nil
 }
 
+// WatchPod is like WatchPods, but for a single key only. The output channel
+// may contain nil manifests, if the target key does not exist.
+func (c consulStore) WatchPod(key string, quitChan <-chan struct{}, errChan chan<- error, podChan chan<- ManifestResult) {
+	defer close(podChan)
+
+	kvpChan := make(chan *api.KVPair)
+	go consulutil.WatchSingle(key, c.client.KV(), kvpChan, quitChan, errChan)
+	for pair := range kvpChan {
+		out := ManifestResult{Path: key}
+		if pair != nil {
+			manifest, err := pods.ManifestFromBytes(pair.Value)
+			if err != nil {
+				select {
+				case <-quitChan:
+					return
+				case errChan <- util.Errorf("Could not parse pod manifest at %s: %s. Content follows: \n%s", pair.Key, err, pair.Value):
```

Comment:
well - the continue would loop back to blocking on kvpChan again. You don't need to select on quit because WatchSingle is doing that for you and will terminate the range loop when it finishes.